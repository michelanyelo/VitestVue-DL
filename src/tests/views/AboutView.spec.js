import { describe, expect, it } from 'vitest'
import { mount } from '@vue/test-utils'
import { createRouter, createWebHistory } from 'vue-router'

import AboutView from '@/views/AboutView.vue'
import App from '@/App.vue'
import router from '@/router'

describe("Renderiza AboutView despuÃ©s de navegar a la ruta 'About'", async () => {
    let wrapper = mount(AboutView)

    const routerPrueba = createRouter({
        history: createWebHistory(),
        routes: [
            {
                path: '/about',
                name: 'about',
                component: AboutView
            }
        ]
    })

    it("Navega y renderiza la vista correspondiente", async () => {
        routerPrueba.push({ name: 'about' })
        await routerPrueba.isReady()

        wrapper = mount(App, {
            global: {
                plugins: [router]
            }
        })

        expect(wrapper.findComponent(AboutView).exists()).toBeTruthy
    })

    // snapshot
    it("El html de AboutView se mantiene en las pruebas", () => {
        expect(wrapper.html()).toMatchSnapshot()
    })
})