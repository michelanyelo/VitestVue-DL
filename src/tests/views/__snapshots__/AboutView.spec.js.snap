// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Renderiza AboutView despuÃ©s de navegar a la ruta 'About' > El html de AboutView se mantiene en las pruebas 1`] = `"<nav><a href="/" class="">Home</a> | <a href="/posts" class="">Posts</a></nav>"`;
